src/components/common/form/FormTextField.tsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react'
+import { TextField } from '@material-ui/core'
+import { FormContextConsumer } from 'components/common/form/FormContext'
+import { translateError } from 'common/form/useForm'
+
+export type RegisterFormProps = {
+  type: string
+  label: string
+  name: string
+  translate: any
+}
```

Comment:
@Nikola-Andreev What's the idea behind passing the `translate` function every time as required prop, instead of using the hook `useTranslation()` inside the component to get it ?

It's much better to strictly type the props, having in mind that we can reuse the `TextFieldProps` straight away from MUI

Reply:
Performance mainly. Why to call `useTranslation()` in each field, when it's already called outside on each form.
But if you think it is better i will add it inside `FormTextField`